### PR TITLE
[navigation]feat: make parent item unclickable and fix duplicate items in landing page.

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description

This PR mainly address 3 issues related to new left navigation.
- Make parent item unclickable in left nav.
- Hide parent item in landing page.
- Filter out hidden nav group when detecting if current application belongs to only one nav group.

### Make parent item unclickable
image

### Only show parent item in landing page and jump to first sub items.
https://github.com/user-attachments/assets/90e50ab9-7795-4336-8324-79413a262526

### Filter out hidden nav group when doing nav group detection (Before fix).
https://github.com/user-attachments/assets/5f5420d0-dbf3-4c62-9d88-adf7b22be1ed

### Filter out hidden nav group when doing nav group detection (After fix).
https://github.com/user-attachments/assets/8ba717f8-53c3-4bc6-832a-e91d15852fe3

## Changelog

- feat: make parent item unclickable and fix duplicate items in landing page.

### Check List

- [x] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled.tsx [hunk 1]
```diff
@@ -135,18 +141,40 @@ export function NavGroups({
...
 
     if (navLink.itemType === LinkItemType.PARENT_LINK && navLink.link) {
-      return {
-        ...createNavItem({ link: navLink.link }),
+      const props = createNavItem({ link: navLink.link });
+      const parentItem = {
+        ...props,
         forceOpen: true,
-        items: navLink.links.map((subNavLink) => createSideNavItem(subNavLink, 'nav-nested-item')),
+        href: undefined,
```
Comment (ruanyl) [inline src/core/public/chrome/ui/header/collapsible_nav_group_enabled.tsx:148]: Nit: maybe leave a comment here to describe why need to set href `undefined` explicitly.

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled.tsx [hunk 2]
```diff
@@ -135,18 +141,40 @@ export function NavGroups({
...
+        onClick: undefined,
+        className: classNames(props.className, 'nav-link-parent-item'),
+        buttonClassName: classNames(props.buttonClassName, 'nav-link-parent-item-button'),
+        items: navLink.links.map((subNavLink) =>
+          createSideNavItem(subNavLink, level + 1, 'nav-nested-item')
+        ),
       };
+      /**
+       * OuiSideBar will never render items of first level as accordion,
+       * in order to display accordion, we need to render a fake parent item.
```
Comment (ruanyl) [inline src/core/public/chrome/ui/header/collapsible_nav_group_enabled.tsx:158]: Is there a screenshot of the accordion?

File: src/core/public/chrome/utils.ts [hunk 3]
```diff
@@ -173,3 +188,28 @@ export function getOrderedLinksOrCategories(
...
+    (props) => {
+      const { currentItem, parentItem } = props;
+      const enricheredResult = enricher ? enricher(currentItem, parentItem) : currentItem;
+      if (enricheredResult.itemType === LinkItemType.LINK) {
+        acc.push(enricheredResult.link);
+      } else if (enricheredResult.itemType === LinkItemType.PARENT_LINK) {
+        if (enricheredResult.link) {
+          acc.push(enricheredResult.link);
+        }
+      }
```
Comment (ruanyl) [inline src/core/public/chrome/utils.ts:211]: Can we simply do this? I'm curious why it doesn't check `enricheredResult.link` for `LinkItemType.LINK`
```suggestion
      if (enricheredResult.itemType === LinkItemType.LINK || enricheredResult.itemType === LinkItemType.PARENT_LINK) {
        if (enricheredResult.link) {
          acc.push(enricheredResult.link);
        }
      }
```

File: src/plugins/management/public/plugin.ts [hunk 4]
```diff
@@ -118,6 +123,61 @@ export class ManagementPlugin implements Plugin<ManagementSetup, ManagementStart
...
+          coreStart.chrome.navLinks.getAll()
+        ).filter((item) => !item.hidden),
+        (currentItem, parentItem) => {
+          /**
+           * Hide sub items in landing page
+           */
+          if (
+            currentItem.itemType === LinkItemType.LINK &&
+            parentItem?.itemType === LinkItemType.PARENT_LINK
+          ) {
```
Comment (ruanyl) [inline src/plugins/management/public/plugin.ts:145]: Q: how do we know this is landing page?

Comment (SuZhou-Joe) [reply src/plugins/management/public/plugin.ts]: The comment may lead to some misunderstanding. Actually it is about we need to hide all the sub items in landing page, and this function will only be called by landing pages to collect all the nav links that await to display.

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled.tsx (hunk 1 above)
Comment (SuZhou-Joe) [reply src/core/public/chrome/ui/header/collapsible_nav_group_enabled.tsx:151]: Sure, added.

File: src/core/public/chrome/utils.ts (hunk 3 above)
Comment (SuZhou-Joe) [reply src/core/public/chrome/utils.ts]: Sure, updated.

File: src/plugins/management/public/plugin.ts (hunk 4 above)
Comment (SuZhou-Joe) [reply src/plugins/management/public/plugin.ts]: ### Before:
image

### After:
image

Comment (SuZhou-Joe) [reply src/plugins/management/public/plugin.ts]: ### Before:
image

### After:
image

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled.tsx (hunk 2 above)
Comment (SuZhou-Joe) [reply src/core/public/chrome/ui/header/collapsible_nav_group_enabled.tsx:161]: ### Before:
image

### After:
image

**Review by ruanyl — APPROVED** [review]

**Review by ananzh — APPROVED** [review]

**Pull request merged**